src/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegment.cs

```diff
@@ -78,6 +78,31 @@ public void ResetMemory()
 
         public Memory<byte> AvailableMemory { get; private set; }
 
+        public IMemoryList<byte> GetNext(long offset, out int localIndex)
```

Comment:
Bounds check on offset?

Reply:
we can add a check for negative here, comparing to `this.RunningIndex` would not be valid.